src/parser/core/modules/Medicated.ts

```diff
@@ -43,22 +43,28 @@ export class Medicated extends Analyser {
 			throw new Error('potion instance not found')
 		}
 
-		this.pots.push({end: event.timestamp, ...this.pot})
+		this.pots.push({ ...this.pot, end: event.timestamp })
 
 		this.pot = undefined
 	}
 
 	private onComplete() {
 		const status = this.data.statuses.MEDICATED
 
+		// Add a pot that is still up before end of pull
+		if (this.pot && !this.pot.end) {
```

Comment:
We require explicit null checks rather than boolean coercion here, consider this instead:
```suggestion
		if (this.pot != null && this.pot.end == null) {
```

For legibility, might be better to make a stopAndSave method and call it in offMedication and here as well. Maybe also close out the window at the start of the onComplete function too.

Reply:
Ahh I referred to other examples of stopAndSave, but is this something you had in mind?
Some files rely on ```this.parser.currentEpochTimestamp``` as the end window time, but most seem to rely on ```this.parser.pull.timestamp + this.parser.pull.duration```. Guessing they'd produce the same value.

```
	private stopAndSave(endTime?: number) {
		if (this.pot != null) {
			this.pots.push({ ...this.pot, end: endTime })
		}
		this.pot = undefined
	}

	private offMedication(event: Events['statusRemove']) {
	        // ...
		this.stopAndSave(event.timestamp);
	}

	private onComplete() {

		// Add a pot that is still up before end of pull
		if (this.pot != null && this.pot.end == null) {
			this.stopAndSave(this.parser.pull.timestamp + this.parser.pull.duration)
		}

               // ...
	}
```

Reply:
The old timestamp format stuff is just from it not being updated I guess. Epoch is the correct way. I'd default the `endTime` param to the current fight time (so you can call it without params if wanted), but basically yes that's the gist of it.

Reply:
Thanks for clarifying. Have opted to rely on Epoch as the default param.